### PR TITLE
add channel slug to SaleorState

### PR DESCRIPTION
**Pull request by AlicjaSzu**

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-614 SDK should keep track of chosen channel](https://app.clickup.com/t/2549495/SALEOR-614)

**Review by orzechdev — CHANGES_REQUESTED** [review]
I'm not sure, but probably we should allow to change channel more dynamically during app usage. Now, when channel in config changes, the SaleorProvider doesn't reflect its change and queries and mutations will be made with previous channel slug passed as variable.

Anyway, I understand that it might require much more changes to sdk and the most important is to support multichannel feature at all for now. So treat it as a sidenote to our future consideration. After few small changes I think it might be merged.

File: src/api/index.ts [hunk 1]
```diff
@@ -74,17 +65,18 @@ export class SaleorAPI {
...
+    this.checkout = new SaleorCheckoutAPI(saleorState, jobsManager, config);
     this.cart = new SaleorCartAPI(
       localStorageManager,
       apolloClientManager,
       saleorState,
-      jobsManager
+      jobsManager,
+      config
     );
     this.categories = new CategoriesAPI(client);
```
[the remark this answers is not in the document]
Comment (jwm0) [reply src/api/index.ts:77]: We don't, categories are not scoped per sales channel

**Comment by jwm0** [comment]
> I'm not sure, but probably we should allow to change channel more dynamically during app usage. Now, when channel in config changes, the SaleorProvider doesn't reflect its change and queries and mutations will be made with previous channel slug passed as variable.

Usually different channels will have different storefront versions (with different env configurations) so this is not a deal breaker right now, but I agree we should look into making this dynamic in future.

**Review by orzechdev — APPROVED** [review]

**Pull request merged**